cmd/istioctl/main.go

```diff
@@ -491,7 +491,7 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(&istioNamespace, "istioNamespace", "i", kube.IstioNamespace,
 		"Istio system namespace")
 
-	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", v1.NamespaceDefault,
+	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", getDefaultNamespace(),
```

Comment:
`getDefaultNamespace()` uses `kubeconfig` but `kubeconfig` is also a flag. `getDefaultNamespace` should be invoked after flags are parsed.